src/AeroSharp/Connection/ClientProvider.cs

```diff
@@ -20,12 +20,13 @@ internal ClientProvider(ConnectionContext connection, Credentials credentials, C
                 user = credentials.Username,
                 password = credentials.Password,
                 asyncMaxCommandAction = GetMaxCommandAction(configuration.MaxCommandAction),
-                asyncMaxCommands = configuration.AsyncMaxCommands
+                asyncMaxCommands = configuration.AsyncMaxCommands,
+                maxConnsPerNode = configuration.MaxConnsPerNode
```

Comment:
Since this is an async policy, we should consider setting asyncMaxConnsPerNode (https://aerospike.com/apidocs/csharp/html/f_aerospike_client_asyncclientpolicy_asyncmaxconnspernode) as noted in the docs for maxConnsPerNode https://aerospike.com/apidocs/csharp/html/f_aerospike_client_clientpolicy_maxconnspernode

Because it's an async policy, it's not clear to me that setting maxConnsPerNode would have much effect.

Reply:
Thank you Austin for reviewing this PR. My response below
* It is true that this is creating an `AsyncClientPolicy` object here, but since `AsyncClientPolicy` inherits from `ClientPolicy`, I would think it is legal to setup a property from parent class when creating a child object like the other properties `user` / `password`?
* We didn't update `asyncMaxConnsPerNode`, and let it default to -1 and use `maxConnsPerNode` according to [here](https://aerospike.com/apidocs/csharp/html/f_aerospike_client_asyncclientpolicy_asyncmaxconnspernode). If we set the default value to 100, then we are changing Aerospike library default behavior (check aerospike source code [here](https://github.com/aerospike/aerospike-client-csharp/blob/master/AerospikeClient/Async/AsyncClientPolicy.cs#L104)) and I would prefer not to do so as a wrapper.
* We have been playing with `maxConnsPerNode` for a while within different teams, and it did impact the behavior a lot.

Please let me know your thought!

Reply:
Ah, I see.

Quoting the relevant portion of that doc here:

> If the value is -1, the value will be set to [maxConnsPerNode](https://aerospike.com/apidocs/csharp/html/f_aerospike_client_clientpolicy_maxconnspernode).

Thanks for following up on this.